exonum/src/helpers/config.rs

```diff
@@ -12,133 +12,13 @@
-    let mut file = File::open(path)?;
-    let mut toml = String::new();
-    file.read_to_string(&mut toml)?;
-    Ok(toml::de::from_str(&toml)?)
-}
-
-fn do_save<T: Serialize>(value: &T, path: &Path) -> Result<(), Error> {
-    if let Some(dir) = path.parent() {
-        fs::create_dir_all(dir)?;
-    }
-    let mut file = File::create(path)?;
-    let value_toml = toml::Value::try_from(value)?;
-    file.write_all(value_toml.to_string().as_bytes())?;
-    Ok(())
-}
-
-/// Structure that handles work with config file at runtime.
-#[derive(Debug)]
-pub struct ConfigManager {
-    handle: thread::JoinHandle<()>,
-    tx: mpsc::Sender<ConfigRequest>,
-}
-
-/// Messages for ConfigManager.
-#[derive(Debug)]
-pub enum ConfigRequest {
-    /// Request for connect list update in config file.
-    UpdateConnectList(ConnectListConfig),
-}
-
-impl ConfigManager {
-    /// Creates a new `ConfigManager` instance for the given path.
-    pub fn new<P>(path: P) -> Self
-    where
-        P: AsRef<Path> + Send + 'static,
-    {
-        let (tx, rx) = mpsc::channel();
-        let handle = thread::spawn(move || {
-            info!("ConfigManager started");
-            for request in rx {
-                match request {
-                    ConfigRequest::UpdateConnectList(connect_list) => {
-                        info!("Updating connect list. New value: {:?}", connect_list);
-
-                        let res = Self::update_connect_list(connect_list, &path);
-
-                        if let Err(ref error) = res {
-                            error!("Unable to update config: {}", error);
-                        }
-                    }
-                }
-            }
-            info!("ConfigManager stopped");
-        });
-
-        ConfigManager { handle, tx }
-    }
-
-    /// Stores updated connect list at file system.
-    pub fn store_connect_list(&self, connect_list: ConnectListConfig) {
-        self.tx
-            .send(ConfigRequest::UpdateConnectList(connect_list))
-            .expect("Can't message to ConfigManager thread");
-    }
-
-    /// Stops `ConfigManager`.
-    pub fn stop(self) {
-        drop(self.tx);
-        self.handle.join().expect("Can't stop thread");
-    }
-
-    // Updates ConnectList on file system synchronously.
-    // This method is public only for testing and should not be used explicitly.
-    #[doc(hidden)]
-    pub fn update_connect_list<P>(connect_list: ConnectListConfig, path: &P) -> Result<(), Error>
-    where
-        P: AsRef<Path>,
-    {
-        let mut current_config: NodeConfig = ConfigFile::load(path)?;
-        current_config.connect_list = connect_list;
-        ConfigFile::save(&current_config, path)?;
-
-        Ok(())
-    }
+/// Interface of the Config Manager usable for updating node configuration on
+/// the fly.
+pub trait ConfigManager: Send {
+    /// Update connect list in the node configuration.
+    fn store_connect_list(&self, connect_list: ConnectListConfig);
```

Comment:
Should the method take `&self` or `&mut self`? The second option is more permissive for implementors (e.g., the list could be stored directly in the manager without needing interior mutability), and, as far as I see, it is allowed by the `NodeHandler` code.